src/main/java/seedu/duke/parser/SearchParser.java

```diff
@@ -0,0 +1,45 @@
+package seedu.duke.parser;
+
+import seedu.duke.command.SearchCommand;
+import seedu.duke.exception.AniException;
+
+public class SearchParser extends CommandParser {
+    protected static final String NAME_PARAM = "n";
+    protected static final String GENRE_PARAM = "g";
+    private SearchCommand searchCommand;
+
+    public SearchParser() {
+        searchCommand = new SearchCommand();
+    }
+
+    public SearchCommand parse(String description) throws AniException {
+        String[] paramGiven = parameterSplitter(description);
+        paramIsSetCheck(paramGiven);
+        parameterParser(paramGiven);
+        return searchCommand;
+    }
+
+    public void parameterParser(String[] paramGiven) throws AniException {
+        for (String param : paramGiven) {
+            String[] paramParts = param.split(" ", 2);
```

Comment:
Introducing a constant here would be good?

Reply:
I have extracted out the values as a constants.